server/src/main/java/com/broadcom/lsp/cobol/core/preprocessor/delegates/util/VariableUtils.java

```diff
@@ -45,8 +45,19 @@
    */
   public @NonNull String createQualifier(
       @NonNull Deque<StructuredVariable> structure, @NonNull String name) {
-    return Optional.ofNullable(structure.peekLast())
-        .map(it -> it.getQualifier() + " ")
-        .orElse("") + name;
+    return Optional.ofNullable(structure.peekFirst())
+            .map(it -> it.getQualifier() + "  ")
+            .orElse(" ")
```

Comment:
Why non-empty string?

Reply:
Here and further, It is for better qualifier concatenation